src/context/campaignConfigsStore.tsx

```diff
@@ -0,0 +1,147 @@
+import React, {
+  createContext,
+  FunctionComponent,
+  useState,
+  useEffect,
+  useCallback
+} from "react";
+import { AsyncStorage } from "react-native";
+import { CampaignConfig } from "../types";
+import { usePrevious } from "../hooks/usePrevious";
+import { CampaignConfigError } from "../services/campaignConfig";
+
+export const CAMPAIGN_CONFIGS_STORE_KEY = "CAMPAIGN_CONFIGS_STORE";
+
+type CampaignConfigsMap = {
+  [key: string]: CampaignConfig | undefined;
+};
+
+export interface CampaignConfigsStoreContext {
+  readonly hasLoadedFromStore: boolean;
+  readonly allCampaignConfigs: CampaignConfigsMap;
+  /**
+   * Key is currently operatorToken + endpoint
+   */
+  addCampaignConfig: (key: string, config: CampaignConfig) => void;
+  removeCampaignConfig: (key: string) => void;
+  clearCampaignConfigs: () => void;
+}
+
+export const CampaignConfigsStoreContext = createContext<
+  CampaignConfigsStoreContext
+>({
+  hasLoadedFromStore: false,
+  allCampaignConfigs: {},
+  addCampaignConfig: () => undefined,
+  removeCampaignConfig: () => undefined,
+  clearCampaignConfigs: () => undefined
+});
+
+export const CampaignConfigsStoreContextProvider: FunctionComponent = ({
+  children
+}) => {
+  const [hasLoadedFromStore, setHasLoadedFromStore] = useState(false);
+  const [allConfigs, setAllConfigs] = useState<
+    CampaignConfigsStoreContext["allCampaignConfigs"]
+  >({});
+
+  const prevAllConfigs = usePrevious(allConfigs);
+
+  useEffect(() => {
+    if (hasLoadedFromStore) {
+      const allConfigsString = JSON.stringify(allConfigs);
+      const prevConfigsString = JSON.stringify(prevAllConfigs);
+      if (allConfigsString !== prevConfigsString) {
+        AsyncStorage.setItem(CAMPAIGN_CONFIGS_STORE_KEY, allConfigsString);
+      }
+    }
+  }, [hasLoadedFromStore, allConfigs, prevAllConfigs]);
+
+  const addCampaignConfig: CampaignConfigsStoreContext["addCampaignConfig"] = useCallback(
+    (key, newConfig) => {
+      if (!newConfig) {
+        return;
+      }
+
+      setAllConfigs(prevConfigs => {
+        const prevConfig: CampaignConfig | undefined = prevConfigs[key];
+        // Tried to use Object.entries to loop rather than directly accessing
+        // as below. But ran into type mismatches.
+        const mergedConfig: CampaignConfig = {
+          features:
+            newConfig.features !== null
+              ? newConfig.features
+              : prevConfig?.features ?? null,
+          policies:
+            newConfig.policies !== null
+              ? newConfig.policies
+              : prevConfig?.policies ?? null
+        };
+
+        return {
+          ...prevConfigs,
+          [key]: mergedConfig
+        };
+      });
+    },
+    []
+  );
+
+  const removeCampaignConfig: CampaignConfigsStoreContext["removeCampaignConfig"] = useCallback(
```

Comment:
hmm just curious. Given that we use `AsyncStorage.getItem`, should we use `AsyncStorage.removeItem` when we have to remove a particular campaign config as well? :O

Reply:
Hmmm we store all the configs under 1 key though, so it isn't really a `removeItem`. Do you think we should split into different keys in async storage?

Reply:
hmm, I think it might be better, i was thinking there's 2 possible benefits:
- if there's any case of malformed data, we only affect one campaign config, instead of everything 
- in the event of a campaign-leak, at least only a single campaign is leaked haha

what do you think? ^^

the cons is definitely harder to query though. 🤔

Reply:
Yea, means that it needs to be async every time, would prefer to put it in "RAM" first, so querying is quick. Also, deletions are easy if we want to flush what's stored on async storage

Regarding the pt on malformed data, we can only control on the backend, unless we want  to validate on frontend too, which I think is not worth the effort. If the user decides to tweak it on his own, he'll reach invalid states and be brought to the error boundary.

Not sure what you mean by campaign leak, we don't encrypt anything haha, so if you compromise async storage, you'll see everything already regardless of key

Reply:
i see i see, that makes sense. hmmm, you are right haha, I blur xD 

i agree with the point on malformed data haha, lets do it the current way then! ^^